_playwright-tests/UI/GPG_keys.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { test, expect } from '@playwright/test';
+import { navigateToRepositories } from './helpers/navHelpers';
+import { deleteAllRepos } from './helpers/deleteRepositories';
+import { closePopupsIfExist, getRowByNameOrUrl } from './helpers/helpers';
+
+export const repoNamePrefix = 'GPG-key';
+export const randomName = () => (Math.random() + 1).toString(36).substring(2, 6);
+export const repoName = `${repoNamePrefix}-${randomName()}`;
+
+export const url = 'https://jlsherrill.fedorapeople.org/fake-repos/signed/';
+export const packages_key =
+  'http://jlsherrill.fedorapeople.org/fake-repos/needed-errata/RPM-GPG-KEY-dummy-packages-generator';
+export const meta_key = 'https://jlsherrill.fedorapeople.org/fake-repos/signed/GPG-KEY.gpg';
+
+test.describe('Test GPG keys', () => {
+  test('Create a repo, add a GPG Key, toggle metadata', async ({ page }) => {
+    await test.step('Delete any GPG key test repos that exist', async () => {
+      await navigateToRepositories(page);
+      await closePopupsIfExist(page);
+      await deleteAllRepos(page, `&search=${repoNamePrefix}`);
+    });
+    await test.step('Create a repository', async () => {
+      // Click on the 'Add repositories' button
+      // HMS-5268 There are two buttons on the ZeroState page
+      await page.getByRole('button', { name: 'Add repositories' }).first().click();
+      await expect(page.getByRole('dialog', { name: 'Add custom repositories' })).toBeVisible();
+      // Fill in the repository details
+      await page.getByLabel('Name').fill(`${repoName}`);
+      await page.getByLabel('Introspect only').click();
+      await page.getByLabel('URL').fill(url);
+      // Paste the URL to the packages GPG key
+      await page.getByPlaceholder('Paste GPG key or URL here').fill(packages_key);
```

Comment:
_small style nit:_ does the comment above this line need to be here? given that we use the user first test actions/locators the actual line it "documents" reads exactly the same as the comment above it, so it doesn't add much value I think, feel free to disagree with me on this 😅 

_(this could be said about more comments in this file)_
_(but if you'd like/prefer to have those comments there feel free to ignore this 😄)_

Reply:
Ok, old habits die hard (using in-line comments). I'll remove it.

Reply:
haha, yeah that's true 😄 there is still place for some, before more actions grouped together when it can be hard to easily figure out what's happening, or ones that link to tickets (like you did in this PR) 😌

thanks for looking into this 🙇🏼‍♂️